src/libpmemobj/palloc.c

```diff
@@ -351,8 +351,16 @@ palloc_reservation_clear(struct palloc_heap *heap,
 		util_mutex_lock(&b->lock);
 		struct memory_block *am = &b->active_memory_block->m;
 
-		if (am->chunk_id == act->m.chunk_id &&
+		/*
+		 * If a memory block used for the action is the currently active
+		 * memory block of the bucket it can be inserted back to the
+		 * bucket. This way it will be easier available in the future.
```

Comment:
This way it will be available for future allocation requests, improving performance.